docs/Architecture/Linters.md

```diff
@@ -52,6 +52,8 @@ These are located in `src/rules/mergers/`, and keyed under their names by the ma
 
 For example, `@typescript-eslint/ban-types` spreads both arguments' `types` members into one large `types` object.
 
+**A merger does not need to be created if the rule does not accept any configuration.**
```

Comment:
A couple nits:
* The docs generally use `> ` quotes for slightly emphasized stuff, to be less emphatic.
* We'll probably end up in this situation:

```suggestion
> A merger does not need to be created if the rule does not accept any configuration, or all converters output the same settings.
```

Reply:
Sure, I'll use `>`, but I think your text modification is wrong.

The merger will not run if two output configuration are the same, but that is just an implementation detail.
A merger needs to exists if there is any configuration at all.

Reply:
If the two converters output the _same_ configuration, we wouldn't want to use a merger. There'd be no need.

Reply:
I understand what you mean, but:

My thought process was that when you add a new converter, then the workflow is:
1. If rule has no configuration: no merger needed. I wrote this in the docs.
2. Else
    1. If other converter(s) output the same rule, then a merger is needed, since you can't assume what configuration arguments will it output and if this arguments output will be the same as the new converter.
    2. Else no merger is needed.

An example for this is `2.i` case is a (fictional) converter like this:

```ts
import { RuleConverter } from "../../ruleConverter";

export const convertNoBigFunction: RuleConverter = (tslintRule) => {
    return {
        rules: [
            {
                ...(tslintRule.ruleArguments.length !== 0 && {
                    ruleArguments: tslintRule.ruleArguments,
                }),
                ruleName: "max-lines-per-function",
            },
        ],
    };
};
```

So I think most if the time *when writing a new converter* you don't know if the output rule arguments will be the same when actually running the program.

So I think you should not encourage not writing a merger when the rule has configuration and there is an existing converter.

Reply:
Maybe it would be better to say that:

> A merger does not need to be created if the rule does not accept any configuration or neither converters outputs arguments.

Reply:
That works for me.

Reply:
I read your suggestion again and I think I misinterpreted it originally. I'll use your suggestion.